### PR TITLE
Issue #10424: Increased Strip Menu Size

### LINKED ISSUE #10424: Default strip menu vertical size is too small to accomodate right hand button
## Description
Default strip menu vertical size is too small to accomodate right hand button. Bit annoying having to resize it every time.

**Reproduction**
- Open strip menu for mob

**Screenshots**
ghost_buckler

### PR DESCRIPTION
**Pull request by wmann95**
Changed the way the strip menu is updated in order to scale based on how many inventory slots are in there, as well as taking into account the size of the buttons.

## About the PR 

**Screenshots**

image

**Changelog**

Fix: Strip menu size wasn't correctly sized for all of the slots.
Added: Strip menu is now dynamically updated based on how many inventory slots there are.

Closes #10424

### CODE REVIEW

**Comment by keronshb** [comment]
As a heads up might want to increase the hight slightly more. With both cuffing and ensnaring, it's going to have two more buttons.

**Comment by EmoGarbage404** [comment]
Why not just change the ui to expand to fit all buttons automatically?

**Comment by EmoGarbage404** [comment]
> > Why not just change the ui to expand to fit all buttons automatically?
> 
> Would be nice but arguably more difficult than just changing hardcoded magic number.

Well obviously but half assing the solution every time instead of actually making it correct is just gonna lead to this problem every few months. Changing the "magic number" isn't a goof solution.

**Comment by wmann95** [comment]
I was trying to do just this, but it's surprisingly hard to just get the actual size required by all of the children, subchildren, etc of the window. I'll play around with it more to see about having it work when the menu is opened, but currently it's changing the positions and scales of all of the children, seen here:

image

**Comment by wmann95** [comment]
Changed this to a draft because there are issues regarding the client side and server side being mixed up somehow. Looking into it.

**Comment by EmoGarbage404** [comment]
just ass a note:
for the changelog, you need to specify it with a prefix (add:, remove:, fix:, tweak:)

You should also put "Closes [issue number here]" in the pr description, which will ensure that the corresponding issue is closed on the pr merge.

File: Content.Client/Inventory/StrippableBoundUserInterface.cs [hunk 1]
```diff
@@ -29,6 +29,7 @@ protected override void Open()
 
             _strippingMenu.OnClose += Close;
             _strippingMenu.OpenCenteredLeft();
+
```
Comment (EmoGarbage404) [inline Content.Client/Inventory/StrippableBoundUserInterface.cs:32]:
```suggestion
```

File: Content.Client/Inventory/StrippableBoundUserInterface.cs [hunk 2]
```diff
@@ -56,6 +57,7 @@ private void UpdateMenu()
                         SendMessage(new StrippingInventoryButtonPressed(slot.ID));
                     });
                 }
+
```
Comment (EmoGarbage404) [inline Content.Client/Inventory/StrippableBoundUserInterface.cs:60]:
```suggestion
```

File: Content.Client/Inventory/StrippableBoundUserInterface.cs [hunk 3]
```diff
@@ -79,6 +81,7 @@ private void UpdateMenu()
                     });
                 }
             }
+
```
Comment (EmoGarbage404) [inline Content.Client/Inventory/StrippableBoundUserInterface.cs:84]:
```suggestion
```

File: Content.Client/Inventory/StrippableBoundUserInterface.cs [hunk 4]
```diff
@@ -91,7 +94,13 @@ protected override void UpdateState(BoundUserInterfaceState state)
             Hands = stripState.Hands;
             Handcuffs = stripState.Handcuffs;
 
+
```
Comment (EmoGarbage404) [inline Content.Client/Inventory/StrippableBoundUserInterface.cs:97]:
```suggestion
```

**Comment by wmann95** [comment]
Gotcha, thank you for the notice. Will update accordingly!

File: Content.Client/Strip/StrippingMenu.cs [hunk 5]
```diff
@@ -59,6 +59,8 @@ public void AddButton(string title, string name, Action<BaseButton.ButtonEventAr
                     button,
                 }
             });
+
+
```
Comment (EmoGarbage404) [inline Content.Client/Strip/StrippingMenu.cs:63]: more unnecessary whitespace.

File: Content.Client/Inventory/StrippableBoundUserInterface.cs [hunk 6]
```diff
@@ -91,7 +94,13 @@ protected override void UpdateState(BoundUserInterfaceState state)
...
             UpdateMenu();
+
+            if (_strippingMenu != null)
+            {
+                _strippingMenu.SetHeight = (Inventory.Count + Hands.Count + Handcuffs.Count) * 34 + _strippingMenu.MinHeight;
```
Comment (EmoGarbage404) [inline Content.Client/Inventory/StrippableBoundUserInterface.cs:102]: i'm a bit sus of magic numbers like this 34. Is there not some property of the button that corresponds to this?

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

File: Content.Client/Inventory/StrippableBoundUserInterface.cs (hunk 6 above)
Comment (wmann95) [reply Content.Client/Inventory/StrippableBoundUserInterface.cs:102]: I'm honestly not sure how best to do this. I've tried getting the height of the _vboxContainer, the height of each of the children of _vboxContainer, the DesiredSize of both, but whenever I do, I get back a height of 0. I assume this is because the button isn't given a Height when it is initialized here:
image
but that said, there might be something I'm missing, as I'm still quite new to the codebase.

Comment (wmann95) [reply Content.Client/Inventory/StrippableBoundUserInterface.cs:102]: I've also looked into the debug locals of the _vboxContainer to find:

image

which shows that almost nothing about the button is actually set. The height is 0, desiredsize is 0, minheight is 0, maxheight is infinity(!),  etc.

**Comment by ElectroJr** [comment]
IMO It's fine to just just hard code the size for now, given that the stripping UI will be overhauled before too long anyways (there is/was already a PR, but its blocked by hud refactor)

File: Content.Client/Inventory/StrippableBoundUserInterface.cs (hunk 6 above)
Comment (ElectroJr) [reply Content.Client/Inventory/StrippableBoundUserInterface.cs]: Given that `IsArangeValid` and `IsMeasureValid` are both false, the issue is probably that you haven't done an initial measure/arrangement yet, and desired size is only set after you've done that.

But see my other comment: IMO hardcoding the UI size is fine for now.

**Comment by wmann95** [comment]
> IMO It's fine to just just hard code the size for now, given that the stripping UI will be overhauled before too long anyways (there is/was already a PR, but its blocked by hud refactor)

As per the last commit, I was able to get it dynamically sized based on the Strip menu children utilizing Measure, which I just learned about. So it isn't hard coded anymore! Unfortunately, had to make use of another loop but since it's only run once when the menu is opened, I think it'll be fine.

File: Content.Client/Strip/StrippingMenu.cs [hunk 7]
```diff
@@ -59,8 +59,19 @@ public void AddButton(string title, string name, Action<BaseButton.ButtonEventAr
...
+        public float GetRequiredHeight()
+        {
+            float sum = 0;
+            for(int i = 0; i < _vboxContainer.ChildCount; i++)
+            {
+                _vboxContainer.GetChild(i).Measure((5000, 5000));
 
+                sum += _vboxContainer.GetChild(i).DesiredSize.Y + _vboxContainer.SeparationOverride ?? 0;
+            }
```
Comment (ElectroJr) [inline Content.Client/Strip/StrippingMenu.cs:72]: you should be able to just measure the vbox directly (which under the hood more or less does the same thing)

Comment (wmann95) [reply Content.Client/Strip/StrippingMenu.cs:72]: You are absolutely correct. I had tried measuring the _strippingMenu but hadn't thought of just doing the _vboxContainer. Thanks for the tip!

**Review by wmann95 — COMMENTED** [review]
I agreed with the removed whitespace.

**Comment by EmoGarbage404** [comment]
did you mean to close this?

**Comment by wmann95** [comment]
I did. I need practice elsewhere before contributing to a project like this, I think. If you want to use this, I can reopen.
